### ISSUE TITLE
[Dynamic] Image brightness filter

### ISSUE THREAD

**Issue by Aqa-Ib** [darkreader/darkreader#509]
In dynamic mode, image brightness is unchanged, and those whose background is white or those with vivid colors, hurts. An image filter like this one is very good for the eyes:

`img {filter:brightness(70%) !important; transition: filter 1s}`
`img:hover, a:hover img  {filter:brightness(85%) !important}`

Regards

**Comment by pdf** [comment]
Would be nice to cover background images too if possible when tackling this (whatever method ends up being used).

**Comment by Gusted** [comment]
#6044

**Comment by r3a1d3a1** [comment]
I haven't tried the latest commit yet, but just wanted to bring up an issue with images that have an alpha channel (translucency). I encounter many of them on Wikipedia. When dark mode is enabled, their background is darkened while the other elements of the image remain as they were. Here's an [example](https://commons.wikimedia.org/wiki/File:Desargues_theorem.svg). It would be best to keep background of such images light and apply a brightness/opacity filter over the whole thing. This way, dark elements of the picture don't become impossible to distinguish. Thanks!

**Comment by Gusted** [comment]
> I haven't tried the latest commit yet, but just wanted to bring up an issue with images that have an alpha channel (translucency). I encounter many of them on Wikipedia. When dark mode is enabled, their background is darkened while the other elements of the image remain as they were. Here's an [example](https://commons.wikimedia.org/wiki/File:Desargues_theorem.svg). It would be best to keep background of such images light and apply a brightness/opacity filter over the whole thing. This way, dark elements of the picture don't become impossible to distinguish. Thanks!

Technically speaking, this isn't possible to fix. There isn't really knowing if a given img element has a alpha channel or not.

**Comment by r3a1d3a1** [comment]
Thank you for responding. 
It's true that a perfect solution in JS for dealing with alpha channels may not be computationally efficient. 
However, I'm currently using an inversion fix (for Filter+) as follows:

```
wikipedia.org

INVERT
.mwe-popups-discreet > svg
.mw-ext-score
.mw-mmv-overlay
#Vorlage_Infobox_Chemikalie > tbody > tr:nth-child(2) > td > a

NO INVERT
.mwe-math-fallback-image-inline
.mwe-math-fallback-image-display
.mwe-popups image
img[alt^="Skeletal" i]
img[alt^="Structural" i]

CSS
:not(.mwe-math-element) > img {
     opacity: 70%
}
```

(Removed the `img[src*="svg.png"]` from NO INVERT section and added the CSS part at the end.)

It does a decent job in most cases. [Example 1](https://en.wikipedia.org/wiki/Thermocouple), [Example 2](https://en.wikipedia.org/wiki/Euclidean_geometry).

**Comment by Gusted** [comment]
> Thank you for responding. It's true that a perfect solution in JS for dealing with alpha channels may not be computationally efficient. However, I'm currently using an inversion fix (for Filter+) as follows:
> 
> ```
> wikipedia.org
> 
> INVERT
> .mwe-popups-discreet > svg
> .mw-ext-score
> .mw-mmv-overlay
> #Vorlage_Infobox_Chemikalie > tbody > tr:nth-child(2) > td > a
> 
> NO INVERT
> .mwe-math-fallback-image-inline
> .mwe-math-fallback-image-display
> .mwe-popups image
> img[alt^="Skeletal" i]
> img[alt^="Structural" i]
> 
> CSS
> :not(.mwe-math-element) > img {
>      opacity: 70%
> }
> ```
> 
> (Removed the `img[src*="svg.png"]` from NO INVERT section and added the CSS part at the end.)
> 
> It does a decent job in most cases. [Example 1](https://en.wikipedia.org/wiki/Thermocouple), [Example 2](https://en.wikipedia.org/wiki/Euclidean_geometry).

It's currently best-effort, but in the future it could be improved e.g. by handling such cases.

**Comment by r3a1d3a1** [comment]
If you like, I can submit a pull request for the inversion fix above.

**Comment by r3a1d3a1** [comment]
Sorry, my bad.

I had been playing around with the inversion fix editor while Dynamic mode was on. Then when I figured out it only affects Filter+, I forgot to check how Wiki looks with default settings on Filter+.
I just reverted my changes and noticed that Wiki on Filter+ looks better than my version.

I wish Filter+ was enabled by default for Wiki, or its Dynamic mode was up to par with its Filter+ version.

Thanks again and sorry for the trouble.
